environments/dev/manifests/loki-query-frontend-service.yaml

```diff
@@ -0,0 +1,23 @@
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    app.kubernetes.io/instance: observatorium-xyz
+    app.kubernetes.io/name: loki
+    app.kubernetes.io/part-of: observatorium
+    app.kubernetes.io/version: 1.5.0
+  name: observatorium-xyz-loki-query-frontend
+  namespace: observatorium
+spec:
+  ports:
+  - name: metrics
+    port: 3100
+    targetPort: 3100
+  - name: grcp
+    port: 9095
```

Comment:
I think you need this one to be `clusterIP: None`. Workers connect to frontend manually via endpoints without using  service IP.

Reply:
AFAIK, i can configure only per service if the cluster IP should be used. In the present configuration a single service is used for both ports. Setting `clusterIP: None` would disable the service IP for `3100` which is used for ingress traffic from other observatorium components, e.g. our observatorium api

Reply:
I think it still fine, the DNS will have 2 IP and the traffic will flow correctly. However having a cluster IP for the GRPC service will cause issue for the frontend workers.

Reply:
```
❯ k describe svc query-frontend -n loki-prod2
Name:              query-frontend
Namespace:         loki-prod2
Labels:            name=query-frontend
Annotations:       kubectl.kubernetes.io/last-applied-configuration:
                     {"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"labels":{"name":"query-frontend"},"name":"query-frontend","namespace":"l...
Selector:          name=query-frontend
Type:              ClusterIP
IP:                None
Port:              query-frontend-http-metrics  80/TCP
TargetPort:        80/TCP
Endpoints:         10.52.215.231:80,10.52.42.238:80
Port:              query-frontend-grpc  9095/TCP
TargetPort:        9095/TCP
Endpoints:         10.52.215.231:9095,10.52.42.238:9095
Session Affinity:  None
Events:            <none>
```

Our gateway also hits the 80 port for incoming **read** requests. For writes our gateway hits directly distributors service using httpgrpc.

```
❯ k describe svc distributor -n loki-prod2
Name:              distributor
Namespace:         loki-prod2
Labels:            name=distributor
Annotations:       kubectl.kubernetes.io/last-applied-configuration:
                     {"apiVersion":"v1","kind":"Service","metadata":{"annotations":{},"labels":{"name":"distributor"},"name":"distributor","namespace":"loki-pr...
Selector:          name=distributor
Type:              ClusterIP
IP:                None
Port:              distributor-http-metrics  80/TCP
TargetPort:        80/TCP
Endpoints:         10.52.13.146:80,10.52.173.28:80,10.52.180.194:80 + 17 more...
Port:              distributor-grpc  9095/TCP
TargetPort:        9095/TCP
Endpoints:         10.52.13.146:9095,10.52.173.28:9095,10.52.180.194:9095 + 17 more...
Session Affinity:  None
Events:            <none>
```

Reply:
For writes you should start with http for now, but I'd recommend to look into httpgrpc later on for high volumes since it will multiplex connections.

Reply:
For the record our offline discussion summarized:
- Need to split the svc between grpc and http.
- GRPC is doing load balancing itself thus we need a headless svc.
- HTTP can benefit from a separate load-balanced svc via clusterIP.

In addition, tailing requests should go to the querier directly.